### ISSUE TITLE
Accessibility Scaling Doesn't Work (Android)

### ISSUE THREAD

**Issue by csnazell** [unoplatform/uno#19056]
### Current behavior

Despite the [documentation](https://platform.uno/docs/articles/feature-flags.html#disabling-accessibility-text-scaling-android-and-ios) stating that Uno automatically scales text in response to accessibility configuration on Android and iOS. This is not happening on either the emulator or test device I am using.

I added the inverse of the disabling code linked above just in case it's now disabled by default to the app constructor to enable text scaling but it had no effect on observed behaviour.

#### At default text size:

![Settings@Default](https://github.com/user-attachments/assets/8d8fd8cd-b048-4cd5-9fc9-0f7c481336c6) 

![AppDraw@Default](https://github.com/user-attachments/assets/0d032c5d-b443-4e6e-99df-71be3e076e76) 

![TestApp@Default](https://github.com/user-attachments/assets/324b406b-6fd2-4ad0-ad07-c9d0a8923cda) 

#### At +2 text size:

![Settings@+2](https://github.com/user-attachments/assets/559c5b35-45e1-4e4e-8dc8-88fb01691de5) 

![AppDraw@+2](https://github.com/user-attachments/assets/6337ae94-5b36-40f8-805e-48fd472c9cb7)

![TestApp@+2](https://github.com/user-attachments/assets/d5f27f9e-38a3-4091-b798-51d07f3cbfa0)

### Expected behavior

Displayed text should change size in response to accessibility configuration made in platform settings.

### Steps to reproduce:

- created a new app from template.

- build & deployed to Android Emulator (API level 35).

- switched to Settings > Display Size and text.

- adjusted Font Size slider to two places larger from the default. 

- switched back to the app. 

- there was no change in the displayed text. 

- switched to app drawer to confirm OS had changed it's default text size. It had.

- deployed to device at API level 33 & it exhibited the same behaviour.

### Demo project 

[UnoAppTest.zip](https://github.com/user-attachments/files/18084987/UnoAppTest.zip)

### Workaround

N/A

### Works on UWP/WinUI

None

### NuGet package version(s)

N/A

### Affected platforms

Android

### IDE

Rider Windows

### IDE version

JetBrains Rider 2024.3 (Build #RD-243.21565.191)

### Relevant plugins

Runtime version: 21.0.5+8-b631.16 amd64 (JCEF 122.1.9)
VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o.
Toolkit: sun.awt.windows.WToolkit
Windows 11.0
.NET Core v8.0.4 x64 (Server GC)
GC: G1 Young Generation, G1 Concurrent GC, G1 Old Generation
Memory: 2048M
Cores: 16
Registry:
  ide.experimental.ui=true
  i18n.locale=
Non-Bundled Plugins:
  com.intellij.resharper.HeapAllocationsViewer (2024.3.0)
  com.jetbrains.rider.android (243.21565.191)
  unoplatform (1.2.28)

### Anything else we need to know?

N / A

**Comment by jeromelaban** [comment]
Thanks for the report. Font scaling is implemented here:
https://github.com/unoplatform/uno/blob/13630a7036172fc59172ce9f865f198c1efaea68/src/Uno.UI/Controls/PaintPool.Android.cs#L153

which gets its values from here:
https://github.com/unoplatform/uno/blob/13630a7036172fc59172ce9f865f198c1efaea68/src/Uno.UI/Extensions/ViewHelper.Android.cs#L86
and
https://github.com/unoplatform/uno/blob/13630a7036172fc59172ce9f865f198c1efaea68/src/Uno.UI/Extensions/ViewHelper.Android.cs#L107

it may be that some android devices don't provide that value properly, or that a change in android broke this feature.

Can you try calling the `ViewHelper.RefreshFontScale()` method to see if this helps? I was able to get the font size to change when restarting the app. It may be that: 

https://github.com/unoplatform/uno/blob/13630a7036172fc59172ce9f865f198c1efaea68/src/Uno.UI/UI/Xaml/ApplicationActivity.Android.cs#L324

does not get called when the system size changes.

**Comment by csnazell** [comment]
I can confirm stopping and restarting the app picks up the new scale value.

I overrode `OnConfigurationChanged()` on the activity as follows:

``` C#
public class MainActivity : Microsoft.UI.Xaml.ApplicationActivity
{
    public override void OnConfigurationChanged(Configuration newConfig)
    {
        base.OnConfigurationChanged(newConfig);
        
        ViewHelper.RefreshFontScale();
    }
}
```

It's definitely getting called by Android (stops at a break point) so `RaiseConfigurationChanges()` should be being called.  Manually calling `ViewHelper.RefreshFontScale()` has no effect.

The newConfig object's .fontScale property reads 1.29999995 at the +2 notch on the scale (cf 1.0 at the default) so this detail is available to the app.

Adding some additional debugging to investigate `RefreshFontScale()`:

``` C#
public class MainActivity : Microsoft.UI.Xaml.ApplicationActivity
{
    public override void OnConfigurationChanged(Configuration newConfig)
    {
        base.OnConfigurationChanged(newConfig);
        
        ViewHelper.RefreshFontScale();

        // display metrics
        
        using Android.Util.DisplayMetrics displayMetrics = Android.App.Application.Context.Resources.DisplayMetrics;
        
        var debugScaledDensity = displayMetrics.ScaledDensity;
        var debugDensity = displayMetrics.Density;

        var debugConfig = FeatureConfiguration.Font.IgnoreTextScaleFactor;
        var debugConfigMaxScale = FeatureConfiguration.Font.MaximumTextScaleFactor;

        return;
    }
}
```

`displayMetrics.Density`           = 2.625

`displayMetrics.ScaledDensity` = 3.4124999

`FeatureConfiguration.Font.IgnoreTextScaleFactor` = false

`FeatureConfiguration.Font.MaximumTextScaleFactor` = null

So `_cachedScaleDensity` should be the same as `displayMetrics.ScaledDensity`

It looks like the UI isn't redrawing itself in response to these changes.

Side note: 

There is a note in the Android docs indicating that the `ScaledDensity` field is deprecated as of API 34 ([here](https://developer.android.com/reference/android/util/DisplayMetrics#scaledDensity)}.

**Comment by MartinZikmund** [comment]
This feature needs to be implemented across all Skia targets
